parsers/rhea/rhea.go

```diff
@@ -0,0 +1,742 @@
+				}
+				newReactionParticipant.Compound = compoundParticipantMap[description.About]
+				rhea.ReactionParticipants = append(rhea.ReactionParticipants, newReactionParticipant)
+			}
+		}
+	}
+	return rhea, nil
+}
+
+// ReadRhea reads in a a gzip'd Rhea dump (https://www.rhea-db.org/help/download) into bytes.
+func ReadRhea(gzipPath string) ([]byte, error) {
+	// Get gz'd file bytes
+	xmlFile, err := os.Open("data/rhea.rdf.gz")
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Decompress gz'd file
+	r, err := gzip.NewReader(xmlFile)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Read decompressed gz'd file
+	rheaBytes, err := ioutil.ReadAll(r)
+	if err != nil {
+		return []byte{}, err
+	}
+	return rheaBytes, nil
+}
+
+/******************************************************************************
+
+Database functions
+
+These functions take in the rhea.rdf.gz dump file and import them into an sqlite
+database. This database can be used locally for mapping functions, but more importantly,
+it is used because it can enforce the relationships between each part of Rhea and alert
+the user if the parser is failing to pick up on any of the relationships in the Rhea
+database.
+
+******************************************************************************/
+
+// CreateRheaSqlite initializes a SQLite database with the proper tables to have Rhea inserted using InsertRheaSqlite.
+func CreateRheaSqlite(db *sql.DB) error {
+	schema := `
+	-- Rhea tables --
+	CREATE TABLE chebi (
+		accession TEXT PRIMARY KEY,
+		subclassof TEXT REFERENCES chebi(accession)
+	);
+	
+	CREATE TABLE compound (
+		id INT,
+		accession TEXT PRIMARY KEY,
+		position TEXT,
+		name TEXT,
+		htmlname TEXT,
+		formula TEXT,
+		charge TEXT,
+		chebi TEXT REFERENCES chebi(accession),
+		polymerizationindex TEXT,
+		compoundtype TEXT NOT NULL CHECK(compoundtype IN ('SmallMolecule', 'Polymer', 'GenericPolypeptide', 'GenericPolynucleotide', 'GenericHeteropolysaccharide'))
+	);
+	
+	CREATE TABLE reactivepart (
+		id INT,
+		accession TEXT PRIMARY KEY,
+		name TEXT,
+		htmlname TEXT,
+		compound TEXT NOT NULL REFERENCES compound(accession)
+	);
+	
+	CREATE TABLE reaction (
+		id INT,
+		directional BOOL NOT NULL DEFAULT false,
+		accession TEXT PRIMARY KEY,
+		status TEXT,
+		comment TEXT,
+		equation TEXT,
+		htmlequation TEXT,
+		ischemicallybalanced BOOL NOT NULL DEFAULT true,
+		istransport BOOL NOT NULL DEFAULT false,
+		ec TEXT,
+		location TEXT
+	);
+	
+	CREATE TABLE reactionside (
+		accession TEXT PRIMARY KEY
+	);
+	
+	CREATE TABLE reactionsidereaction (
+		reaction TEXT NOT NULL REFERENCES reaction(accession),
+		reactionside TEXT NOT NULL REFERENCES reactionside(accession),
+		substrateorproduct BOOL NOT NULL DEFAULT false, -- If set to true, disregard substrate below
+	        substrate BOOL NOT NULL DEFAULT false -- If substrateOrProduct is false, "substrate = true" means this is a substrate and "substrate = false" means this is a product
```

Comment:
Seems like this is a actually an "enum":

`reactionsidereactiontype TEXT NOT NULL CHECK(reactionsidereactiontype IN ('substrateorproduct', 'substrate', 'product'))`